crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -288,15 +291,19 @@ impl Entity {
     ///
     /// This method is the fallible counterpart to [`Entity::from_bits`].
     #[inline(always)]
-    pub const fn try_from_bits(bits: u64) -> Result<Self, &'static str> {
-        if let Some(generation) = NonZeroU32::new((bits >> 32) as u32) {
-            Ok(Self {
-                generation,
-                index: bits as u32,
-            })
-        } else {
-            Err("Invalid generation bits")
+    pub const fn try_from_bits(bits: u64) -> Result<Self, IdentifierError> {
+        if let Ok(id) = Identifier::try_from_bits(bits) {
+            let kind = id.kind() as u8;
```

Comment:
Are these conversion to `u8` necessary?

Reply:
Yes, because otherwise the comparison is not `const`.